src/components/UserRegisterPage.vue

```diff
@@ -162,26 +219,107 @@ export default {
       firstname: '',
       lastname: '',
       displayName: 'Researcher',
+      colors: Array(),
+      pattern: Array(),
     },
+    isCreated: false,
+    isLoading: false,
     nda: null,
     repeatPassword: '',
     errors: [],
+    palettes:
+    [
+      '#96ceb4', '#ffeead', '#ffcc5c', '#ff6f69', '#588c7e', '#f2e394', '#f2ae72', '#d96459',
+      '#a9bdc8', '#677e96', '#4a9bb1', '#ccd6e6', '#4f615b', '#3d95a6', '#d3deec', '#3c4b54',
+      '#3e8696', '#dce5f4', '#45535f', '#4a818a', '#b2bdcc', '#2e4051', '#62797d'
+    ],
+    numColors: 5,
   }),
   components: {
     ValidationProvider,
     ValidationObserver,
   },
+  computed: {
+    userLabel() {
+      if (this.user.firstname.length || this.user.lastname.length) {
+        return `${this.user.firstname} ${this.user.lastname}`;
+      }
+      return this.$t('signUp');
+    },
+    patternAsText: {
+      get() {
+        if (this.user) {
+          return this.user.pattern.join(', ');
+        }
+        return '';
+      },
+      set(v) {
+        this.user.setPattern(v);
+      }
+    },
+  },
   methods: {
     onSubmit() {
-      console.info('UserPage.onSubmit()', this.user, this.nda);
+      // console.info('UserPage.onSubmit()', this.user, this.nda);
       // to be checked for validity...
-      this.$store.dispatch('user/CREATE_USER_ACCOUNT', this.user, this.nda);
+      this.isLoading = true;
+      usersService.create(this.user)
+        .then(() => {
+          this.isLoading = false;
```

Comment:
It's better to move `this.isLoading = false` into `finally(...)`. Right now if an error occurs it is logged to console and the loading indicator is never turned off.

Reply:
done